lib/cdo/data/logging/test/test_rake_event_logger.rb

```diff
@@ -0,0 +1,46 @@
+require lib_dir 'cdo/data/logging/rake_task_event_logger'
+
+require_relative '../../../../../lib/test/test_helper'
+require 'rake'
+
+include TimedTaskWithLogging
+
+class RakeEventLoggerTest < ActiveSupport::TestCase
+  def test_failed_task
+    RakeTaskEventLogger.any_instance.expects(:start_task_logging).once
+    RakeTaskEventLogger.any_instance.expects(:end_task_logging).never
+    RakeTaskEventLogger.any_instance.expects(:exception_task_logging).once
+    error = assert_raises(RuntimeError) do
+      Rake::Task['test_exception_timed_task'].invoke
+    end
+    assert_equal 'This is really bad', error.message
+  end
+
+  def test_successful_task
+    RakeTaskEventLogger.any_instance.expects(:start_task_logging).once
+    RakeTaskEventLogger.any_instance.expects(:end_task_logging).once
+    RakeTaskEventLogger.any_instance.expects(:exception_task_logging).never
+    Rake::Task['test_successful_timed_task'].invoke
+  end
+
+  def test_failed_logger_task
+    RakeTaskEventLogger.any_instance.expects(:start_task_logging).once
+    RakeTaskEventLogger.any_instance.expects(:end_task_logging).once
+    RakeTaskEventLogger.any_instance.expects(:exception_task_logging).never
+    RakeTaskEventLogger.any_instance.stubs(:log_event).raises(RuntimeError, "failed logger")
+    Rake::Task['test_failed_logger_timed_task'].invoke
+    RakeTaskEventLogger.any_instance.unstub(:start_task_logging)
+  end
```

Comment:
This is more a pet peeve of mine around testing. This test case describes the **setup** scenario of a failed logger, but it doesn't describe what the **expected** behavior is unless you carefully read the code.

If I'm understanding this, the expected behavior is that the Task will complete successfully without raising any exceptions, even if the logger throws an error. I'm inferring this because you don't wrap the Task call in a handler like you do in the "failed task" test case.

That's why I like to see descriptive test names, something like `test_failed_logger_does_not_fail_task`

also... I guess now I'm seeing the answer to my other question. This is a manual testing script, not an automated test script. Can we get this into the automated test suite?

Reply:
```
This is a manual testing script, not an automated test script. Can we get this into the automated test suite?
```

How? :)

Reply:
gonna checked it in, it might be I got the wrong folder? I'll check in, check what's going on in drone and if it is not there, I'll move it around on a new PR

Reply:
I'll check in once staging reopens so I can get a drone run for the other 2 PRs I have connected to this one